src/Admin/AdminHelper.php

```diff
@@ -248,23 +248,29 @@ public function addNewInstance($object, FieldDescriptionInterface $fieldDescript
 
         $method = sprintf('add%s', Inflector::classify($mapping['fieldName']));
 
-        if (!\is_callable([$object, $method])) {
-            $method = rtrim($method, 's');
+        if (!method_exists($object, $method)) {
+            $nonSMethod = rtrim($method, 's');
 
-            if (!\is_callable([$object, $method])) {
-                $method = sprintf('add%s', Inflector::classify(Inflector::singularize($mapping['fieldName'])));
-
-                if (!\is_callable([$object, $method])) {
-                    /*
-                     * NEXT_MAJOR: Use BadMethodCallException instead
-                     */
-                    throw new \RuntimeException(
-                        sprintf('Method %s::%s() does not exist.', ClassUtils::getClass($object), $method)
-                    );
+            if (method_exists($object, $nonSMethod)) {
+                $method = $nonSMethod;
+            } else {
+                $singularizeMethod = sprintf('add%s', Inflector::classify(Inflector::singularize($mapping['fieldName'])));
+
+                if (method_exists($object, $singularizeMethod)) {
+                    $method = $singularizeMethod;
                 }
             }
         }
 
+        if (!\is_callable([$object, $method])) {
```

Comment:
If you have a protected nonSMethod exist and a public singularizeMethod you'll get an error.

Best is just to replace all `\is_callable` by `method_exist && is_callable`.